tests/regtest.py

```diff
@@ -0,0 +1,2002 @@
+				if path != testpath:
+					self.assertMode(path, mode)
+				else:
+					self.assertMode(path, self.m_modes[profile][mode_index])
+
+			print()
+
+class TestCapabilities(TestBase):
+
+	def __init__(self):
+
+		super().__init__("TestCapabilities", "checks whether capability settings and related command line options work")
+
+	def run(self):
+
+		if not self.m_main_test_instance.haveCapSupport():
+			self.printWarning("Cannot set file capabilities in user namespaces on this kernel. It only works starting from version 4.14")
+			return
+
+		testfile = "/caps_test"
+		self.createTestFile(testfile, 0o755)
+
+		# just test a single profile in this case, we just want to see
+		# whether caps work at all
+		profile = "easy"
+		mode = 0o750
+		caps = ["cap_net_admin", "cap_net_raw=ep"]
+
+		entries = {
+			profile: [ self.buildProfileLine(testfile, mode, caps = caps) ]
+		}
+
+		self.addProfileEntries(entries)
+
+		self.switchSystemProfile(profile)
+		# by default caps should be set, if in the sysconfig
+		# configuration not value is set (but the variable still needs
+		# to be there).
+		self.applySystemProfile()
+
+		self.assertHasCaps(testfile, caps)
+
+		os.unlink(testfile)
+		self.createTestFile(testfile, 0o755)
+		self.applySystemProfile(["--no-fscaps"])
+
+		# this time there should be no extended attribute at all
+		self.assertNoCaps(testfile)
+
+class TestUnexpectedPathOwner(TestBase):
+
+	def __init__(self):
+
+		super().__init__("TestUnexpectedPathOwner", "checks whether changes are rejected when parent dir owner and target path owner don't match")
+
+	def run(self):
+
+		testdir = self.createAndGetTestDir(0o755)
+		baddir = os.path.join( testdir, "dir" )
+		badfile = os.path.join( testdir, "file" )
+
+		self.createTestFile(badfile, 0o644)
+		self.createTestDir(baddir, 0o755)
+
+		testprofile = "easy"
+
+		entries = {
+			testprofile: (
+				# add a trailing slash to express that we want
+				# a directory here
+				self.buildProfileLine(baddir + "/", 0o500),
+				self.buildProfileLine(badfile, 0o600)
+			)
+		}
+
+		self.addProfileEntries(entries)
+		# for creating the mixed ownership we need to resort to bind
+		# mounts, since we're not really root and can't chown()
+		# (at least not without sub*id support)
+		self.m_main_test_instance.bindMount("/bin/bash", badfile)
+		self.m_main_test_instance.bindMount("/usr/bin", baddir)
+		orig_file_mode = self.getMode(badfile)
+		orig_dir_mode = self.getMode(baddir)
+
+		try:
+			self.switchSystemProfile(testprofile)
+			code, lines = self.applySystemProfile()
+			# make sure modes actually didn't change
+			# before bind mounts are removed
+			self.assertMode(badfile, orig_file_mode)
+			self.assertMode(baddir, orig_dir_mode)
+		finally:
+			self.m_main_test_instance.unmount(badfile)
+			self.m_main_test_instance.unmount(baddir)
+
+		found_dir_reject = False
+		found_file_reject = False
+
+		# we can't evaluate the exit code, it's quite always zero, due
+		# to RPM packaging requirements.
```

Comment:
But it's not? The RPM scriptlet (in factory) has `|| true` to ignore the exit code, but `chkstat` does do a `exit(1)` when errors were encountered.

If you see something different, it's something is broken, or maybe the `unshare` etc. hide the real exit code?

Reply:
Hmm, the reasoning I added there was wrong, but for the problem at hand it still proves true:

```
/TestUnexpectedPathOwner/file: has unexpected owner. refusing to correct due to unknown integrity.
```

This situation doesn't cause a non-zero exit code. As various kinds of `safe_open()` errors don't trigger a bad exit code. Therefore I've just adjusted the comment for the moment.